WooCommerce/Classes/ViewRelated/JetpackSetup/JetpackSetupCoordinator.swift

```diff
@@ -0,0 +1,89 @@
+import UIKit
+import Yosemite
+import enum Alamofire.AFError
+
+/// Coordinates the Jetpack setup flow in the authenticated state.
+///
+final class JetpackSetupCoordinator {
+    let navigationController: UINavigationController
+
+    private let site: Site
+    /// Whether Jetpack is installed and activated and only connection needs to be handled.
+    private var requiresConnectionOnly: Bool
+    private let stores: StoresManager
+    private let analytics: Analytics
+
+    init(site: Site,
+         navigationController: UINavigationController,
+         stores: StoresManager = ServiceLocator.stores,
+         analytics: Analytics = ServiceLocator.analytics) {
+        self.site = site
+        self.requiresConnectionOnly = false // to be updated later after fetching Jetpack status
+        self.navigationController = navigationController
+        self.stores = stores
+        self.analytics = analytics
+    }
+
+    func showBenefitModal() {
+        let benefitsController = JetpackBenefitsHostingController(isJetpackCPSite: site.isJetpackCPConnected)
+        benefitsController.setActions (installAction: { [weak self] result in
+            guard let self else { return }
+            self.analytics.track(event: .jetpackInstallButtonTapped(source: .benefitsModal))
+            if let result, self.site.isNonJetpackSite {
+                self.handleJetpackStatus(result)
+            } else {
+                self.presentJCPJetpackInstallFlow()
+            }
+        }, dismissAction: { [weak self] in
+            self?.navigationController.dismiss(animated: true, completion: nil)
+        })
+        navigationController.present(benefitsController, animated: true, completion: nil)
+    }
+}
+
+// MARK: - Private helpers
+//
+private extension JetpackSetupCoordinator {
+    /// Navigates to the Jetpack installation flow for JCP sites.
+    func presentJCPJetpackInstallFlow() {
+        navigationController.dismiss(animated: true, completion: { [weak self] in
+            guard let self else { return }
+            let installController = JCPJetpackInstallHostingController(siteID: self.site.siteID,
+                                                                       siteURL: self.site.url,
+                                                                       siteAdminURL: self.site.adminURL)
+
+            installController.setDismissAction { [weak self] in
+                self?.navigationController.dismiss(animated: true, completion: nil)
+            }
+            self.navigationController.present(installController, animated: true, completion: nil)
+        })
+    }
+
+    /// Checks the Jetpack connection status for non-Jetpack sites to infer the setup steps to be handled.
+    func handleJetpackStatus(_ result: Result<JetpackUser, Error>) {
```

Comment:
Super nit: `handleJetpackStatus` to me feels like the method receives the Jetpack status as a parameter and handles it. 
If you think it makes sense kindly consider using `checkJetpackStatus`.

Reply:
Done in 841c511bfc0e8ed4adce28c2fd44ee5f3fe6139a.